File: Backend/SpringBoot/HospitalManagementApi/src/main/java/com/example/demo/entity/Doctor.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "doctor")
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int doctor_id;

    private String dfName;
    private String dlName;
    private String mobileNo;
    private String emailId;
    private String dAddress;
    private String registrationId;

    @Temporal(TemporalType.DATE)
    private Date doj;

    private String specialization;

    @ManyToOne
    @JoinColumn(name = "dept_id")
    @JsonIgnoreProperties({"doctors"})
    private Department department;

    @ManyToOne
    @JoinColumn(name = "loginId")
    @JsonIgnoreProperties({"doctor"})
    private Login login;

   
}

File: Backend/Dot Net/LoginWebRestApi/LoginWebRestApi/Models/Doctor.cs
using System;
using System.Collections.Generic;

namespace LoginWebRestApi.Models;

public partial class Doctor
{
    public int DoctorId { get; set; }

    public string DfName { get; set; } = null!;

    public string? DlName { get; set; }

    public string MobileNo { get; set; } = null!;

    public string EmailId { get; set; } = null!;

    public string DAddress { get; set; } = null!;

    public string RegistrationId { get; set; } = null!;

    public DateOnly Doj { get; set; }

    public string Specialization { get; set; } = null!;

    public int DeptId { get; set; }

    public int LoginId { get; set; }

    public virtual Department Dept { get; set; } = null!;

    public virtual Login Login { get; set; } = null!;
}
